sdk/core/Azure.Core.Experimental/src/JsonData.cs

```diff
@@ -20,15 +18,204 @@ namespace Azure.Core.Dynamic
     /// <summary>
     /// A mutable representation of a JSON value.
     /// </summary>
-    [DebuggerDisplay("{DebuggerDisplay,nq}")]
-    [DebuggerTypeProxy(typeof(JsonDataDebuggerProxy))]
+    //[DebuggerDisplay("{DebuggerDisplay,nq}")]
+    //[DebuggerTypeProxy(typeof(JsonDataDebuggerProxy))]
     [JsonConverter(typeof(JsonConverter))]
     public partial class JsonData : DynamicData, IDynamicMetaObjectProvider, IEquatable<JsonData>
     {
-        private readonly JsonValueKind _kind;
-        private Dictionary<string, JsonData>? _objectRepresentation;
-        private List<JsonData>? _arrayRepresentation;
-        private object? _value;
+        private Memory<byte> _original;
+        private List<IJsonDataChange>? _changes;
+        private JsonElement _element;
+
+        internal JsonDataElement RootElement => new JsonDataElement(this, _element, "");
+
+        internal bool HasChanges => _changes != null && _changes.Count > 0;
+
+        internal bool TryGetChange<T>(string path, out T? value)
+        {
+            if (_changes == null)
+            {
+                value = default(T);
+                return false;
+            }
+
+            for (int i = _changes.Count - 1; i >= 0; i--)
+            {
+                var change = _changes[i];
+                if (change.Property == path)
+                {
+                    // TODO: does this do boxing?
+                    value = ((JsonDataChange<T>)change).Value;
```

Comment:
This is unboxing. The list is of interface instances, i.e. the list stores boxed values. This is not ideal. Why do you need this interface?